Sources/Controllers/Cells/OAFolderCardsCell.m

```diff
@@ -72,6 +74,35 @@ - (void) setValues:(NSArray<NSString *> *)values sizes:(NSArray<NSNumber *> *)si
     [self.collectionView reloadData];
 }
 
+- (void) scrollToItem:(NSInteger)selectedIndex
+{
+    if (_cachedSelectedItemIndex != selectedIndex)
+    {
+        [self.collectionView layoutIfNeeded];
+        [self.collectionView scrollToItemAtIndexPath:[NSIndexPath indexPathForItem:selectedIndex inSection:0]
+                                    atScrollPosition:UICollectionViewScrollPositionCenteredHorizontally
+                                            animated:YES];
+        _cachedSelectedItemIndex = selectedIndex;
+    }
+}
+
+- (void) updateVisibleItem:(NSInteger)index
+{
+    for (NSInteger i = 0; i < _data.count; i++)
+    {
+        NSDictionary *item = _data[i];
+        _data[i] = @{
+            @"title" : item[@"title"],
+            @"size" : item[@"size"],
+            @"color" : item[@"color"],
+            @"img" : item[@"img"],
+            @"isSelected" : [NSNumber numberWithBool:i == index],
```

Comment:
Rather than storing the selected state in the immutable collection, it is marginally better to set the selection state for each cell based on the currently selected item variable.